### ISSUE TITLE
Update dask.dataframe.map_partitions docs to match dask.dataframe.DataFrame.map_partitions

### ISSUE THREAD

**Issue by fbunt** [dask/dask#9223]
When I look up the documentation for `map_partitions`, I use the search box on the docs page. The top result is usually `dask.dataframe.map_partitions` so that is what I click on and that is the documentation I use as I assume that it is the same or similar to the docs for the method. Recently I wanted to see if map_partitions allows for a function argument like block_info. There is no mention of this on the docs page that I normally go to. I just went to `dask.dataframe.DataFrame.map_partitions` for the first time and I was really surprised by all of the extra information, including mention of a `partition_info` arg.

It seems like the module level function's docs are out of date compared to the `DataFrame` method's. It would be great if the docs were updated considering that the module level function's appear first in in-site searches.

**Comment by fbunt** [comment]
@faulaire Yep!

**Comment by faulaire** [comment]
> @faulaire Yep!

Sorry, I removed my reply because I saw that you were talking about an issue on map_partitions and I updated map_overlap documentation in #9145.

So, the issue you are talking about is not resolved.

**Comment by jsignell** [comment]
Thanks for bring this up! It seems like there is an opportunity to consolidate a lot of docs :) 

I am trying to think what the general solution should be. I looked to numpy to see how they handle this and they basically only have module docs. The method docs immediately link out the the module.

 - [module func docs](https://numpy.org/doc/stable/reference/generated/numpy.mean.html)
 - [method docs](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.mean.html)

So that is one option, remove all the method docs and absorb them into the module docs. 

There is another option which is to programmatically create the method docs _from_ the module docs. We do this kind of thing already for dask methods that are based on other libraries (like numpy or pandas) we use `derived_from` to copy the docs from one place to the other. That seems mildly preferable to me since it would rely on a single source of truth, but have the information wherever the user looks for it rather than making them follow links. 

What do you think?

**Comment by fbunt** [comment]
Yeah the numpy style is what I'm used to. I just checked pandas and they do the latter, however. I guess it comes down to what you all, as the maintainers, find easiest.
